packages/components/pie-toast/src/index.ts

```diff
@@ -64,15 +64,48 @@ export class PieToast extends RtlMixin(LitElement) implements ToastProps {
     @property({ type: Object })
     public leadingAction: ToastProps['leadingAction'];
 
+    @property({ type: Number })
+    public duration = defaultProps.duration;
+
     @query('pie-button') actionButton?: HTMLElement;
 
+    @query('pie-icon-button') closeButton?: HTMLElement;
+
     private _actionButtonOffset = 0;
 
     private _messageAreaMaxWidth = 0;
 
+    private _timeoutId: NodeJS.Timeout | null = null;
+
+    private _abortController: AbortController | null = null;
+
     // Renders a `CSSResult` generated from SCSS by Vite
     static styles = unsafeCSS(styles);
 
+    /**
+     * Create a timeout function and set its id into a private attribute.
+     *
+     * @private
+     */
+    private setAutoDismiss () {
+        this._timeoutId = setTimeout(() => {
+            this.closeToastComponent();
+        }, this.duration || 0 as never);
```

Comment:
Does this mean that if duration is passed as null, it will fall back to 0 and therefore execute immediately? I know we have `as never` but that seems like something we either shouldn't need to specify, or should actively prevent.

Reply:
If the duration is null this method will NEVER be executed. Since TS complains if I don't provide a fallback here, I am adding zero as never since zero should never be a value here.

Reply:
I think this might be another way around it?
```typescript
private setAutoDismiss () {
    if (this.duration === null) return;

    this._timeoutId = setTimeout(() => {
        this.closeToastComponent();
    }, this.duration);
}
```